src/lsrs/helper/tlp-transport.ts

```diff
@@ -0,0 +1,109 @@
+import axios from "axios";
+import {ILogObject, TTransportLogger} from "tslog/src/interfaces";
+
+/**
+ * Domain class representing an error object in the log.
+ */
+class ErrorLog {
+
+    private readonly className: string;
+    private readonly message: string;
+    private readonly stackTrace?: string;
+
+    constructor(error: Error) {
+        this.className = error.constructor.name;
+        this.message = error.message;
+        this.stackTrace = typeof error.stack == "string"
+            ? error.stack
+            : JSON.stringify(error.stack);
+    }
+}
+
+/**
+ * Domain class representing a TLP API compatible log message.
+ */
+class TLPLogMessage {
+
+    private static readonly _SOURCE_APPLICATION = "lsrs";
+
+    private readonly source: string;
+    private readonly threadName: string;
+    private readonly timeStamp: number;
+    private readonly loggerName: string;
+    private readonly level: {levelStr: string};
+    private readonly content: string;
+    private readonly exception?: ErrorLog;
+    
+    /**
+     * This constructor extract and converts log entry data from a TSLog log entry object.
+     *
+     * @param entry TSLog log entry object of type ILogObject
+     */
+    constructor(entry: ILogObject) {
+
+        this.source = TLPLogMessage._SOURCE_APPLICATION;
+        this.threadName = entry.requestId || "main";
+        this.timeStamp = entry.date.getTime();
+        this.loggerName = entry.loggerName ?? "default";
+        this.level = {levelStr: entry.logLevel.toUpperCase()};
+        this.content = entry.argumentsArray[0] as string;
```

Comment:
Is this guaranteed to be always a string? Would it make sense to use `toString()`?

Reply:
Out of pure habit, I'm always passing a string as the first parameter of a log, so yes, it'll always be a string.